hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/Associator.java

```diff
@@ -65,7 +65,7 @@ public Relationship associationWith(Path path) throws RelationNotFoundException
     }
 
     protected void checkPathLegal(Path targetPath) {
-        if (!context.entityClass.equals(targetPath.getSegment().getElementType())) {
+        if (!context.entityClass.getSimpleName().equals(targetPath.getSegment().getElementType().getSimpleName())) {
```

Comment:
Not sure if I like string matching here. What creating some helper in inventory API, say `InventoryPaths`, which would convert this in an ever so slightly efficient manner? Maybe a `toSegmentType(Class)` method using some `IdentityHashMap` (or actually a series of `if`s) and `toEntityType(SegmentType)` being actually just your `Entity.entityTypeFromSegmentType()`? 

Or maybe even have a dedicate method for this - `InventoryPaths.representsSegmentType(Class, SegmentType)` which could be made using a switch and equality comparison on a class (which is just reference equality and thus very quick).